### ISSUE TITLE
@next/mdx not working with remarkPlugins

### ISSUE THREAD

**Issue by alfredoalpizar** [vercel/next.js#39590]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

    Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Pro
    Binaries:
      Node: 16.15.0
      npm: N/A
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 12.2.5
      eslint-config-next: 12.2.5
      react: 18.2.0
      react-dom: 18.2.0

### Describe the Bug

When using a remarkPlugin I created and manipulating the unified AST tree, @next/mdx doesn't process the javascript in the updated AST.

if logic inside the remarkPlugin adds:
```js
export const meta = {"title":"test"}
```
to the tree, then @next/mdx should still process that and it shouldn't appear on the generated page

### Expected Behavior

the JS code that gets added in the remarkPlugin should still be handled by @next/mdx and not appear in the generated page.

for example if index.mdx actually had the same JS code in its file like this:
```mdx
# My MDX Page with a Layout

This is a list in markdown:

- One
- Two
- Three

Checkout my React component:
export const meta = {"title":"test"}
```
then the last export line wouldn't show up in the generated page because @next/mdx would process it.

### Link to reproduction

https://stackblitz.com/edit/vercel-next-js-jb1ebi?file=next.config.js

### To Reproduce

example:
```js
// next.config.js
const frontmatterPlugin = require('./plugins/frontmatterPlugin.js')

/** @type {import('next').NextConfig} */
const withMDX = require('@next/mdx')({
  extension: /\.mdx?$/,
  options: {
    remarkPlugins: [frontmatterPlugin],
    // If you use `MDXProvider`, uncomment the following line.
    // providerImportSource: "@mdx-js/react",
  },
})

const nextConfig = withMDX({
  reactStrictMode: true,
  swcMinify: true,
  // Append the default value with md extensions
  pageExtensions: ['ts', 'tsx', 'js', 'jsx', 'md', 'mdx'],
})

module.exports = nextConfig
```

```js
// frontmatterPlugin.js
const fm = require('gray-matter');

module.exports = () => (tree, file) => {
    const {content, data} = fm(file.value)
    
    // adding JS code to .mdx file being processed
    tree.children.push({
        type: 'export',
        value: `export const meta = ${JSON.stringify(data)}`,
    })
    
    if(tree.children[0].type === 'thematicBreak') {
        const firstHeadingIndex = tree.children.findIndex(t => t.type === 'heading')
        if (firstHeadingIndex !== -1) {
          tree.children.splice(0, firstHeadingIndex + 1)
        }
    }
}
```

index.mdx:
```mdx
---
title: test
---
# My MDX Page with a Layout

This is a list in markdown:

- One
- Two
- Three

Checkout my React component:
```

generated page output:
![image](https://user-images.githubusercontent.com/20887630/184545871-3b8fae6b-5a27-46b7-826f-332afab7d93a.png)

**Comment by alfredoalpizar** [comment]
Maybe I'm not using the remarkPlugins option as intended?

If so I apologize for the confusion but I couldn't find anything in the docs or online that give more info

**Comment by alfredoalpizar** [comment]
my issue was that I wasn't pushing stuff to the unifiedjs tree correctly. 
```js
    tree.children.push({
        type: 'export',
        value: `export const meta = ${JSON.stringify(data)}`,
    })
```
was incorrect because the type should have been `mdxjsEsm` and it was missing additional info in the object I was pushing.

I took a stab at adding stuff to the AST correctly but I decided it was too cumbersome to do it this way. 
instead I opted for using a custom webpack loader with @next/mdx that adds the frontmatter data and my layout import/export before its processed by unifiedjs which created the AST afterwards.
